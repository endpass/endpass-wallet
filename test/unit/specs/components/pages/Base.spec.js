import { shallowMount } from '@vue/test-utils';
import VeeValidate from 'vee-validate';
import BasePage from '@/components/pages/Base';

describe('Base page', () => {
  let wrapper;
  let options;

  beforeEach(() => {
    options = {
      provide: () => ({
        $validator: new VeeValidate.Validator(),
      }),
    };

    wrapper = shallowMount(BasePage, options);
  });

  describe('render', () => {
    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('BasePage');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render the initial state of the page', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render default slot', () => {
      const wrapper = shallowMount(BasePage, {
        ...options,
        slots: {
          default: '<div>default slot</div>',
        },
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render title slot', () => {
      const wrapper = shallowMount(BasePage, {
        ...options,
        slots: {
          title: 'title',
        },
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
