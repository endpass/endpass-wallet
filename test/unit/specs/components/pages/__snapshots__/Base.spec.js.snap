// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base page render should render default slot 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          />
           
          <a
            class="card-header-icon"
          >
            
            &lt; Back
          
          </a>
        </div>
         
        <div
          class="card-content"
        >
          <div>
            default slot
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Base page render should render the initial state of the page 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          />
           
          <a
            class="card-header-icon"
          >
            
            &lt; Back
          
          </a>
        </div>
         
        <div
          class="card-content"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Base page render should render title slot 1`] = `
<div
  class="app-page"
>
  <div
    class="section"
  >
    <div
      class="container is-narrow"
    >
      <div
        class="card app-card main-app-card"
      >
        <div
          class="card-header"
        >
          <h1
            class="card-header-title"
          >
            title
          </h1>
           
          <a
            class="card-header-icon"
          >
            
            &lt; Back
          
          </a>
        </div>
         
        <div
          class="card-content"
        />
      </div>
    </div>
  </div>
</div>
`;
